omim2obo/main.py

```diff
@@ -164,63 +164,69 @@ def omim2obo(use_cache: bool = False):
+        elif is_gene:  # * or +
             graph.add((omim_uri, RDFS.subClassOf, SO['0000704']))  # gene
             graph.add((omim_uri, MONDO.exclusionReason, MONDO.nonDisease))
             graph.add((omim_uri, BIOLINK['category'], BIOLINK['Gene']))
         elif omim_type == OmimType.PHENOTYPE:
             graph.add((omim_uri, BIOLINK['category'], BIOLINK['Disease']))  # phenotype ~= disease
         elif omim_type == OmimType.SUSPECTED:
             graph.add((omim_uri, MONDO.exclusionReason, MONDO.excludeTrait))
-        else:
-            pass
 
-        if use_abbrev_over_label and abbrev:
-            graph.add((omim_uri, RDFS.label, Literal(abbrev)))
+        # Alternative rdfs:label for genes
+        if is_gene and pref_symbols:
+            gene_label_err = 'Warning: Only 1 symbol picked for label for gene term, but there were 2 to choose' \
+                 f'from. Unsure which is best. Picking the first.\nhttps://omim.org/entry/{omim_id} - {pref_symbols}'
+            if len(pref_symbols) > 1:
+                LOG.warning(gene_label_err)  # todo: decide the best way to handle these situations
+            graph.add((omim_uri, RDFS.label, Literal(pref_symbols[0])))
         else:
-            graph.add((omim_uri, RDFS.label, Literal(label_cleaner.clean(label))))
-
-        # Reify on abbreviations. See: https://github.com/monarch-initiative/omim/issues/2
-        exact_labels = [s.strip() for s in label.split(';')]
-        if len(exact_labels) > 1:
-            abbr = exact_labels.pop()
-            graph.add((omim_uri, oboInOwl.hasExactSynonym, Literal(abbr)))
-            # Acronym: uppercase, no numbers, no whitespace, <10 chars
-            if abbr.isupper() and not any(char.isspace() for char in abbr) and len(abbr) < 10:
-                axiom = BNode()
-                graph.add((axiom, RDF.type, OWL.Axiom))
-                graph.add((axiom, OWL.annotatedSource, omim_uri))
-                graph.add((axiom, OWL.annotatedProperty, oboInOwl.hasExactSynonym))
-                graph.add((axiom, OWL.annotatedTarget, Literal(abbr)))
-                graph.add((axiom, oboInOwl.hasExactSynonym, MONDONS.abbreviation))
-        for exact_label in exact_labels:
-            graph.add((omim_uri, oboInOwl.hasExactSynonym, Literal(label_cleaner.clean(exact_label, abbrev))))
-        for label in other_labels:
-            graph.add((omim_uri, oboInOwl.hasExactSynonym, Literal(label_cleaner.clean(label, abbrev))))
+            graph.add((omim_uri, RDFS.label, Literal(label_cleaner.clean(pref_title))))
+
+        # todo: .clean() should accept all symbols/abbrevs (preferred, alt, included), else [] instead of str/None
+        #  - See: https://github.com/monarch-initiative/omim/issues/129
+        abbrev: Union[str, None] = None if not pref_symbols else pref_symbols[0]
+
+        # Add synonyms
+        graph.add((omim_uri, oboInOwl.hasExactSynonym, Literal(label_cleaner.clean(pref_title, abbrev))))
```

Comment:
This line looks new since the [last commit we reviewed](https://github.com/monarch-initiative/omim/pull/130/commits/d41df383cee4656c176b0905e2650185319b263b). Do you have the diff of the omim.owl files or the .ttl file that is created with this newer version of the code?

Reply:
Very observant. This line was there when I ran it with you yesterday and looked at the diff. Pretty sure it had been pushed by that time, but probably not when you reviewed. If it was missing when you reviewed, that was a mistake; it should've never have been missing.

This operation isn't new per se. It is a refactor of:

https://github.com/monarch-initiative/omim/blob/db4ddd2e88a773d1c5743c82529c665edc1a925c/omim2obo/main.py#L210-L211

Previously, the code had referred to "preferred title" as "exact labels". IDK why. It also assumed there could be more than 1 preferred title, which there can't. So IIRC, this is also part of what was causing lowercasing in one of the symbols for [those 2 symbol cases](https://github.com/monarch-initiative/omim/issues/125)

Reply:
Also FYI, this 1 line is the topic of this issue as well:
- #131

Reply:
> would like to see the new .ttl file

I undeleted the [google drive folder](https://drive.google.com/drive/folders/1yeQAfh_yzI8vCEzkaBblF6-B__EEJFsT) w/ the uploads we were looking yesterday, reran and created a new `omim.ttl`, and uploaded it there (renamed to `after3.ttl`).

I looked at the `md5`s, but indeed as we were discussing yesterday, it is non-deterministic, and even when running it back-to-back, I got a different hash each time. Comparing `.ofn` would be more ideal, but it looks like I deleted the one we were looking at yesterday (directly from my IDE, so it's not in my trash either).

Reply:
Thanks for the updated file, the abbreviations look good.